### PR TITLE
Add optional AES-256 encryption for file storage

### PR DESCRIPTION
**Pull request by bplatz**
## Summary
Adds optional AES-256 encryption support for file storage in Fluree DB, providing data-at-rest encryption for sensitive ledger data.

## Key Features
- **Optional encryption**: Backward compatible, encryption only enabled when `aes256-key` is provided
- **AES-256-CBC encryption**: Industry standard encryption with PKCS5 padding
- **Transparent operation**: Encryption/decryption happens automatically during file I/O
- **Secure key management**: Support for environment variables and JSON-LD configuration
- **Performance optimized**: Minimal overhead (~10-20% CPU impact)

### New Parameter: `aes256-key`
```clojure
;; Basic file storage (unchanged)
@(fluree/connect-file {:storage-path "./data"})

;; File storage with encryption (new)
@(fluree/connect-file {:storage-path "./data"
                       :aes256-key "my-secret-32-byte-encryption-key\!"})
```

## Implementation Details
- **Core changes**: Updated `FileStore` record to handle encryption transparently
- **System integration**: Added `AES256Key` vocabulary and system configuration support
- **Security**: Uses fluree.crypto library for AES-256-CBC with proper IV handling
- **Compatibility**: Existing ledgers continue to work without encryption

## Documentation
- **API documentation**: Added comprehensive docstring to `connect-file` function
- **Configuration guide**: New `docs/FILE_STORAGE_GUIDE.md` with examples and best practices
- **Quick start**: Updated README.md with encryption examples

## Testing
- **Comprehensive test suite**: 4 new test scenarios covering encryption functionality
- **Edge cases**: Tests wrong key handling, raw file verification, content-addressable storage
- **Integration**: Verifies encryption key propagation from API to FileStore
- **All tests pass**: 207/207 tests passing, no regressions

## Security Considerations
- **Key requirements**: 32-byte keys for optimal AES-256 security
- **Environment variables**: Recommended for production key management
- **Data migration**: Encryption cannot be enabled on existing ledgers (new ledgers only)
- **Key loss**: Data cannot be recovered without the correct encryption key

## Files Changed
- `src/fluree/db/api.cljc`: Add aes256-key parameter and documentation
- `src/fluree/db/storage/file.cljc`: Implement encryption in FileStore
- `src/fluree/db/connection/vocab.cljc`: Add AES256Key vocabulary
- `src/fluree/db/connection/system.cljc`: Pass encryption key to FileStore
- `docs/FILE_STORAGE_GUIDE.md`: Complete configuration guide
- `test/fluree/db/storage/file_test.clj`: Comprehensive test suite
- `README.md`: Add Quick Start examples with encryption

## Test Results
```
207 tests, 1489 assertions, 3 pending, 0 failures
```

Ready for review and integration.

### CODE REVIEW

File: src/fluree/db/storage/file.cljc [hunk 1]
```diff
@@ -51,32 +73,61 @@
...
-        (full-path path)
-        fs/read-file)))
+    (go-try
+      #?(:clj
+         ;; For Clojure, we need to read raw bytes
+         (let [file-path (full-path root path)]
+           (try
+             (let [file (java.io.File. file-path)]
+               (when (.exists file)
+                 (let [raw-bytes (java.nio.file.Files/readAllBytes (.toPath file))]
```
Comment (zonotope) [inline src/fluree/db/storage/file.cljc]: Instead of this deeply nested structure, I think this code could be improved with a few file utility functions.
- something like `(file-exists? path)` could take a string path and report if a file exists at that path.
- something like `(read-bytes path)` could take a string path and read the bytes stored there.
- a third function combining the two that automatically returns `nil` if no file exists at that path, and the stored bytes if the file does exist.

Comment (bplatz) [reply src/fluree/db/storage/file.cljc]: I refactored it a bit to keep the file storage protocol implementation clean and move the conditional logic into the util file which I think addresses your concern and feels more reasonable

**Review by zonotope — APPROVED** [review]
🛌🏾
